pkg/manifests/config.go

```diff
@@ -266,26 +266,39 @@ func (c *Config) applyDefaults() {
 		}
 	}
 
-	if c.ClusterMonitoringConfiguration.MetricsServerConfig == nil {
+	// TODO(slashpai): Remove adapter check in this after 4.16
```

Comment:
I would keep some flexibility around the removal. In practice, I expect that we keep it for at least 2 minor releases (4.17 + 4.18) and we block upgrades in 4.18 if the entry is still present.